src/plumtree_broadcast.erl

```diff
@@ -321,12 +345,14 @@ handle_cast({update, Members}, State=#state{all_members=BroadcastMembers,
 %% @private
 -spec handle_info('exchange_tick' | 'lazy_tick' | {'DOWN', _, 'process', _, _}, state()) ->
     {noreply, state()}.
-handle_info(lazy_tick, State) ->
-    schedule_lazy_tick(),
+handle_info(lazy_tick,
+            #state{lazy_tick_period = Period} = State) ->
+    schedule_lazy_tick(Period),
     _ = send_lazy(State),
     {noreply, State};
-handle_info(exchange_tick, State) ->
-    schedule_exchange_tick(),
+handle_info(exchange_tick,
+            #state{exchange_tick_period = Period} = State) ->
+    schedule_exchange_tick(Period),
```

Comment:
Schedule should happen after the work is done, to prevent them from backing up, no?